diff --git a/facets.install b/facets.install
index d95e9eb..8480fab 100644
--- a/facets.install
+++ b/facets.install
@@ -30,8 +30,11 @@ function facets_update_8001() {
   foreach ($entities as $entity) {
     $facetSourceId = $entity->getFacetSourceId();
     foreach ($old_ids as $id) {
-      if (strpos($facetSourceId, $id) !== FALSE) {
-        $new_id = str_replace($id . ':', 'search_api:' . $id . '__', $facetSourceId);
+      $replacement = 'search_api:' . $id . '__';
+      // Replace if the id follows the old pattern and it does not look like
+      // we've already made the replacement.
+      if (strpos($facetSourceId, $id) !== FALSE && strpos($facetSourceId, $replacement) === FALSE) {
+        $new_id = str_replace($id . ':', $replacement, $facetSourceId);
         $entity->setFacetSourceId($new_id);
         $entity->save();
       }
